c/cert/src/rules/ENV31-C/EnvPointerIsInvalidAfterCertainOperations.ql

```diff
@@ -0,0 +1,46 @@
+/**
+ * @id c/cert/env-pointer-is-invalid-after-certain-operations
+ * @name ENV31-C: Do not rely on an env pointer following an operation that may invalidate it
+ * @description EnvironmentPointerIsInvalidAfterCertainOperations
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/env31-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+
+/*
+ * Call to functions that modify the environment.
+ */
+
+class EnvModifyingCall extends FunctionCall {
+  EnvModifyingCall() { this.getTarget().hasGlobalOrStdName(["setenv", "_putenv_s"]) }
```

Comment:
also, do we care about any other env modification mechanisms? is `putenv` relevant? is it possible to reassign the env pointer to a non const declaration then change the env that way?